ArduCopter/control_guided.cpp

```diff
@@ -212,6 +213,31 @@ void Copter::guided_set_angle(const Quaternion &q, float climb_rate_cms)
     guided_angle_state.update_time_ms = millis();
 }
 
+// guided_reached_destination_notify - notify if we've reached the destination
+// on guided mode - if we're taking off we consider altitude
+void Copter::guided_reached_destination_notify(bool check_altitude)
+{
+    bool reached_alt = true;
+
+    if (guided_prev_mode != guided_mode) {
+        guided_reached_notified = false;
+    }
+
+    if (check_altitude) {
+        const Vector3f& curr_pos = inertial_nav.get_position();
+        const Vector3f& target_pos = wp_nav.get_wp_destination();
+
+        if (curr_pos.z < target_pos.z) {
+            reached_alt = false;
```

Comment:
Maybe, since it's a very dated patch I should review it all over. I remember of may discussions on the overall approach and the overall requirements vs. desired features. Are we really interested on this right now? I also remember on some issues regarding different vehicles implementation.